lib/tasks/fastboot-server.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const RSVP = require('rsvp');
+const CoreObject = require('core-object');
+const debug = require('debug')('ember-cli-fastboot/server');
+const exec = RSVP.denodeify(require('child_process').exec);
+const http = require('http');
+const path = require('path');
+
+module.exports = CoreObject.extend({
+
+  exec,
+  http,
+  httpServer: null,
+  nextSocketId: 0,
+  require,
+  restartAgain: false,
+  restartPromise: null,
+  sockets: {},
+
+  run(options) {
+    debug('run');
+    const restart = () => this.restart(options);
+    process.on('SIGHUP', restart);
+  },
+
+  start(options) {
+    debug('start');
+    this.ui.writeLine('Installing FastBoot npm dependencies');
+
+    return this.exec('npm install', { cwd: options.outputPath })
+      .then(() => {
+        const middleware = this.require('fastboot-express-middleware')(options.outputPath);
```

Comment:
We should change how we invoke `fastboot-express-middleware` to make reloading easier.  We should basically copy from how it is done in `fastboot-app-server`:

* https://github.com/ember-fastboot/fastboot-app-server/blob/master/lib/worker.js#L52-L61 - Setting up the middleware.
* https://github.com/ember-fastboot/fastboot-app-server/blob/master/lib/worker.js#L41 - To reload/restart.